### ISSUE TITLE
php 8.4 ,  multiple 'nullable type' deprecation notices

### ISSUE THREAD

**Issue by pgnd** [handcraftedinthealps/ZendSearch#9]
install of
```
composer show | grep -i zend
	handcraftedinthealps/zendsearch          2.1.0             a general purpose text search engine wr...
```

with
```
php -v
	PHP 8.4.1 (cli) (built: Nov 20 2024 09:48:35) (NTS gcc x86_64)
	Copyright (c) The PHP Group
	Built by Remi's RPM repository <https://rpms.remirepo.net/> #StandWithUkraine
	Zend Engine v4.4.1, Copyright (c) Zend Technologies
	    with Zend OPcache v8.4.1, Copyright (c), by Zend Technologies
	    with Xdebug v3.4.0, Copyright (c) 2002-2024, by Derick Rethans
```

exec reports multiple deprecation notices:
```
Deprecated: ZendSearch\Lucene\Index\SegmentInfo::termDocs(): Implicitly marking parameter $docsFilter as nullable is deprecated, the explicit nullable type must be used instead in /srv/sulu.app/vendor/handcraftedinthealps/zendsearch/library/ZendSearch/Lucene/Index/SegmentInfo.php on line 918

Deprecated: ZendSearch\Lucene\Index\SegmentInfo::termFreqs(): Implicitly marking parameter $docsFilter as nullable is deprecated, the explicit nullable type must be used instead in /srv/sulu.app/vendor/handcraftedinthealps/zendsearch/library/ZendSearch/Lucene/Index/SegmentInfo.php on line 1034

Deprecated: ZendSearch\Lucene\Index\SegmentInfo::termPositions(): Implicitly marking parameter $docsFilter as nullable is deprecated, the explicit nullable type must be used instead in /srv/sulu.app/vendor/handcraftedinthealps/zendsearch/library/ZendSearch/Lucene/Index/SegmentInfo.php on line 1154
```

**Comment by alexander-schranz** [comment]
@pgnd looks like any easy fix. Do you want to create a pull request for it?

**Comment by pgnd** [comment]
-> https://github.com/handcraftedinthealps/ZendSearch/pull/10

**Comment by alexander-schranz** [comment]
Released as [2.2.0](https://github.com/handcraftedinthealps/ZendSearch/releases/tag/2.2.0)
